### PR TITLE
Add relative date filters UI

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/34112

How to verify:
- Orders -> Filter -> Created At -> Relative dates
- It should be possible to set past/current/next options
- Check `Starting from` and `Include current` options

Screenshot 2023-10-06 at 15 03 42
Screenshot 2023-10-06 at 15 03 27
Screenshot 2023-10-06 at 23 18 53
Screenshot 2023-10-06 at 23 14 15

### CODE REVIEW

File: frontend/src/metabase/ui/components/inputs/NumberInput/NumberInput.styled.tsx [hunk 1]
```diff
@@ -22,9 +22,6 @@ export const getNumberInputOverrides =
         params: NumberInputStylesParams,
         { size = "md" }: ContextStylesParams,
       ) => ({
-        wrapper: {
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/NumberInput/NumberInput.styled.tsx:25]: This doesn't work when there is no label

File: frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/RelativeDatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  const [value, setValue] = useState(initialValue ?? DEFAULT_VALUE);
+  const direction = getDirection(value);
+  const isNew = initialValue == null;
+
+  const handleTabChange = (tabValue: string | null) => {
+    const tab = TABS.find(tab => tab.direction === tabValue);
+    if (tab) {
+      setValue(setDirection(value, tab.direction));
+    }
+  };
```
Comment (kulyk) [inline frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/RelativeDatePicker.tsx:39]: Is `tabValue` expected to be `null`?

File: frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/constants.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export const TABS: Tab[] = [
+  { label: t`Past`, direction: "last" },
+  { label: t`Current`, direction: "current" },
+  { label: t`Next`, direction: "next" },
+];
+
+export const UNIT_GROUPS: DatePickerTruncationUnit[][] = [
+  ["day", "week", "month"],
+  ["quarter", "year"],
+];
```
Comment (kulyk) [inline frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/constants.ts:14]: This one doesn't seem to be used anywhere. `CurrentDatePicker` has its own definition of `UNIT_GROUPS` (in the constants) file

File: frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/types.ts
Comment (kulyk) [inline frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/types.ts:1]: Do we need this file? It's empty

**Review by kulyk — APPROVED** [review]
Looks amazing 🏆 Left a few minor-ish comments, please take a look before merging

File: frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/types.ts
Comment (iethree) [reply frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/types.ts:1]: oh, this seems like it's what is making typescript throw a fit

File: frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/RelativeDatePicker.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/RelativeDatePicker.tsx:39]: https://mantine.dev/core/tabs/#tab-deactivation

By default not but the type requires `null`

**Pull request merged**